public/components/report_definitions/report_settings/report_settings.tsx

```diff
@@ -138,46 +138,63 @@ function TimeRangeSelect() {
   );
 }
 
-export function ReportSettings() {
+export function ReportSettings(props) {
+  const { createReportDefinitionRequest, httpClientProps } = props;
+
   const [reportName, setReportName] = useState('');
   const [reportDescription, setReportDescription] = useState('');
   const [reportSourceId, setReportSourceId] = useState('dashboardReportSource');
-  const [dashboardSourceSelect, setDashboardSourceSelect] = useState(
-    REPORT_SOURCE_DASHBOARD_OPTIONS[0].value
-  );
+
+  const [dashboardSourceSelect, setDashboardSourceSelect] = useState('');
+  const [dashboards, setDashboards] = useState([]);
+
   const [visualizationSourceSelect, setVisualizationSourceSelect] = useState(
     REPORT_SOURCE_VISUALIZATION_OPTIONS[0].value
   );
   const [savedSearchSourceSelect, setSavedSearchSourceSelect] = useState(
     REPORT_SOURCE_SAVED_SEARCH_OPTIONS[0].value
   );
   const [fileFormat, setFileFormat] = useState('pdfFormat');
+
   const [savedSearchFileFormat, setSavedSearchFileFormat] = useState(
     'csvFormat'
   );
   const [includeHeader, setIncludeHeader] = useState(false);
   const [includeFooter, setIncludeFooter] = useState(false);
 
+  const handleDashboards = (e) => {
+    setDashboards(e);
+  };
+
   const handleReportName = (e: {
     target: { value: React.SetStateAction<string> };
   }) => {
     setReportName(e.target.value);
+    createReportDefinitionRequest['report_name'] = e.target.value.toString();
   };
 
   const handleReportDescription = (e: {
     target: { value: React.SetStateAction<string> };
   }) => {
     setReportDescription(e.target.value);
+    createReportDefinitionRequest['description'] = e.target.value.toString();
   };
 
   const handleReportSource = (e: React.SetStateAction<string>) => {
     setReportSourceId(e);
+    if (e === 'dashboardReportSource') {
+      createReportDefinitionRequest['report_source'] = 'Dashboard';
+    } else if (e === 'visualizationReportSource') {
+      createReportDefinitionRequest['report_source'] = 'Visualization';
+    }
   };
 
   const handleDashboardSelect = (e: {
     target: { value: React.SetStateAction<string> };
   }) => {
     setDashboardSourceSelect(e.target.value);
+    createReportDefinitionRequest['report_params']['url'] = 
```

Comment:
Why are these still being used as `[]` instead of dot notation.

Reply:
I thought that was in reference to the http response? `createReportDefinitionRequest` is controlled by us so isn't there not the same risk of running into `undefined`?